### PR TITLE
fix fbxloader error

### PR DESCRIPTION
**Pull request by FishOrBear**
[Drawing1.zip](https://github.com/mrdoob/three.js/files/1225547/Drawing1.zip)
Fixed an error.
The error caused fbx not to be imported properly.

#9860

### CODE REVIEW

File: examples/js/loaders/FBXLoader.js [hunk 1]
```diff
@@ -4010,7 +4010,10 @@
 			while ( ! this.endOfContent( reader ) ) {
 
 				var node = this.parseNode( reader, version );
-				if ( node !== null ) allNodes.add( node.name, node );
+				if ( node !== null )
```
Comment (mrdoob) [inline examples/js/loaders/FBXLoader.js]: How about reversing it?

```js
if ( node === null ) break;
allNodes.add( node.name, node );
```

**Comment by takahirox** [comment]
Let me review if this fix is correct.

File: examples/js/loaders/FBXLoader.js (hunk 1 above)
Comment (FishOrBear) [inline examples/js/loaders/FBXLoader.js]: nice.
Do i need to modify pr?

Comment (FishOrBear) [inline examples/js/loaders/FBXLoader.js]: I have modified pr

**Comment by takahirox** [comment]
It doesn't seem this change is correct.
With Drawing1.fbx, this change makes `FBXLoader` break a loop at a certain point in the binary file but some data still exist after the point.
This means `FBXLoader` loses some data.

Probably "Unknown property type" isn't a root issue.
There seems being another root issue which causes it.
I'm taking a look closer...

**Comment by FishOrBear** [comment]
3dmax before exporting
image
After import
[test.zip](https://github.com/mrdoob/three.js/files/1226960/test.zip)

image

There is also a problem that causes the imported grid to appear incorrectly.

[Uploading testBox.zip…]()


@takahirox

**Comment by takahirox** [comment]
Thanks for reporting. Would you please check if there's any existing similar/same issue threads and open another issue for that if there isn't?

**Comment by FishOrBear** [comment]
@takahirox  
Find 2 related questions.

#11697  #11895

I copied the wrong instance file to  #11895

**Comment by takahirox** [comment]
If I'm right, Drawing1.fbx includes two models...?
How was this file generated?

**Comment by FishOrBear** [comment]
@takahirox 
This is the AutoCAD2016 derived fbx file.
If I quit in the middle, then the entity is loaded normally.

**Comment by takahirox** [comment]
> If I quit in the middle

What do you mean by that?

**Comment by FishOrBear** [comment]
Sorry, this is the result of Google translation, may be somewhat misleading.

That is, if i follow my code.
It can load the model correctly.

That is to say

If (null === node) // exit. (If I quit in the middle) (google translation error)


@takahirox

**Comment by takahirox** [comment]
OK. Let me explain what I mentioned again.

FBX binary has 16-byte end-footer in the end of the file.

    0xf8 0x5a 0x8c 0x6a 0xde 0xf5 0xd9 0x7e 0xec 0xe9 0x0c 0xe3 0x75 0x8f 0x29 0x0b

But Drawing1.fbx has two footers, at address 0x4160 and 0x4440.
I suppose Drawing1.fbx includes two FBX data.
Probably you can load without any `FBXLoader` change if you cut the second data.

I'm really not sure if this "two (or more) combined data in one file" is proper FBX specification.

IMO We can consider to support it if
- There're (not a few) this type of existing files
- Other libs/viewers can load and render them correctly

**Comment by FishOrBear** [comment]
This file is imported into u3d, or UE4, which can be imported correctly.

I do not know how to split the file.

**Comment by takahirox** [comment]
Can you generate the ascii version file and share?
I'm wondering how we should handle the second data.
(Just ignore?)

**Comment by FishOrBear** [comment]
@takahirox 
I do not know how to generate ascii files.
I think i can try fbxSDK

**Comment by FishOrBear** [comment]
[Drawing1Ascii.zip](https://github.com/mrdoob/three.js/files/1236390/Drawing1Ascii.zip)
I tried using fbxConver to convert the file.


http://usa.autodesk.com/adsk/servlet/pc/item?siteID=123112&id=22694909

**Comment by looeee** [comment]
@FishOrBear @takahirox  I've tested loading the original file (Drawing1.fbx) in a couple of ways, including importing it back into 3ds Max. In all cases it doesn't look like the original screenshot - it's just six planes making up a box. 

Additionally as you can see in the following screen shot, the planes are missing names - this is unusual (actually first time I've seen this in max, generally everything is named). 

box


Could it be that the original export didn't work properly for some reason?

**Comment by FishOrBear** [comment]
@looeee  
I guess this is not the reason.
You can try the following file.

[Drawing4.zip](https://github.com/mrdoob/three.js/files/1236821/Drawing4.zip)

**Comment by takahirox** [comment]
Anyways, I want something proof which indicates this type of files is a proper FBX format. I'd never seen.

**Comment by mrdoob** [comment]
This got messed up...

**Comment by FishOrBear** [comment]
@takahirox  Hi,After many years, I continued the discussion of this issue.

https://github.com/mont29/blender-io-fbx/blob/master/io_scene_fbx/parse_fbx.py#L166

I think my conjecture is right, because the code of blender is also like this.

**Comment by FishOrBear** [comment]
https://github.com/mrdoob/three.js/blob/dev/examples/js/loaders/FBXLoader.js#L1697

Actually in my file, this property may be null, so I am modifying it so
```
if (node.attrName)
    model.name = THREE.PropertyBinding.sanitizeNodeName(node.attrName);
```

**Pull request closed without merging**